src/ng/browser.js

```diff
@@ -235,10 +235,22 @@ function Browser(window, document, $log, $sniffer) {
 
   // This variable should be used *only* inside the cacheState function.
   var lastCachedState = null;
+  function getCurrentState() {
+    try {
+      var state = history.state;
+      if (state === void 0) {
```

Comment:
Looks like we are pretending that it will never be `undefined` : https://github.com/angular/angular.js/blob/master/src/ng/browser.js#L159-L164